src/transformers/models/nucleus_x/configuration_nucleus_x.py

```diff
@@ -0,0 +1,211 @@
+# This code was adapted from Sehyun Choi's https://github.com/syncdoth/retnet
+# licensed under the MIT License. Above repository builds on Microsoft's
+# torchscale library and the RetNet implementations in this library, also
+# licensed under the MIT License. It has been modified from its original forms
+# to accommodate minor architectural differences compared to NucleusX used by
+# the NucleusAI team that trained the model.
+
+# MIT License
+#
+# Copyright (c) 2023  NucleusAI and The HuggingFace Inc. team and Sehyun Choi
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+""" NucleusX model configuration"""
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+NUCLEUS_X_PRETRAINED_CONFIG_ARCHIVE_MAP = {}
+
+
+class NucleusXConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`~NucleusXModel`]. It is used to instantiate an
+    NucleusX model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the NucleusX-7B
+    [NucleusAI/NucleusX-7B](https://huggingface.co/NucleusAI/NucleusX-7B) architecture.
```

Comment:
This weight is not released yet; we are planning to release the weight at this link soon! We have included this link to pass the configuration testing requiring a link to the checkpoint.